Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -94,10 +106,14 @@ class EdgeHitProcessor: HitProcessing {
     /// - Parameters:
     ///   - entityId: unique id of the `DataEntity` used when the hit needs to be retried.
     ///   - event: event to be processed.
+    ///   - eventConfig: configuration data for this event.
     ///   - requestBuilder: the `RequestBuilder` object to build the request payload.
     ///   - completion: completion handler to notify the caller about the hit response.
-    private func processExperienceEvent(entityId: String, event: Event, requestBuilder: RequestBuilder, completion: @escaping (Bool) -> Void) {
-        guard let edgeConfig = getEdgeConfig(event: event), var datastreamId = edgeConfig[EdgeConstants.SharedState.Configuration.CONFIG_ID] else {
+    private func processExperienceEvent(entityId: String, event: Event, edgeConfig: [String: String], requestBuilder: RequestBuilder, completion: @escaping (Bool) -> Void) {
+        guard var datastreamId = edgeConfig[EdgeConstants.SharedState.Configuration.CONFIG_ID], !datastreamId.isEmpty else {
+            Log.warning(label: EdgeConstants.LOG_TAG,
+                        "\(SELF_TAG) - Unable to process the event '\(event.id.uuidString)' " +
+                            "due to invalid edge.configId in configuration.")
```

Comment:
- rewording - let's clarify what is the exact invalid scenario here: due to invalid ... -> due to missing/empty ...
- I assume this will only be possible for the old path since this verification can be done prior to queueing. could we 1. ensure queued config is valid 2. do this check for old workflow and finalize the config validity even before calling these methods

Reply:
The check on configId is only needed for the old path, as the configId is validated to be non-nil/non-empty before queuing the data entity. However, I still need to extract the configId in processExperienceEvent/processConsentEvents so it's natural to use a `guard` for this and simple enough to include the `!isEmpty` check. If I were to perform the check earlier, I'd still need to unwrap the `configId` from the EdgeDataEntity, so I'd still need to include a `guard` or some other such statement.